src/contracts/facilitators/gsm/feeStrategy/GsmFeeStrategyFactory.sol

```diff
@@ -0,0 +1,84 @@
+/// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {EnumerableSet} from '@openzeppelin/contracts/utils/structs/EnumerableSet.sol';
+import {VersionedInitializable} from '@aave/core-v3/contracts/protocol/libraries/aave-upgradeability/VersionedInitializable.sol';
+import {IGsmFeeStrategyFactory} from './interfaces/IGsmFeeStrategyFactory.sol';
+import {IGsmFeeStrategy} from 'src/contracts/facilitators/gsm/feeStrategy/interfaces/IGsmFeeStrategy.sol';
+import {FixedFeeStrategy} from './FixedFeeStrategy.sol';
+
+/**
+ * @title GsmFeeStrategyFactory
+ * @author Aave Labs
+ * @notice Factory contract to create and keep record of Gsm Fee contracts
+ */
+contract GsmFeeStrategyFactory is VersionedInitializable, IGsmFeeStrategyFactory {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  mapping(uint256 => mapping(uint256 => address)) internal _gsmFeeStrategiesByRates;
```

Comment:
add docs here please

Reply:
Done - Although happy to change to something else